src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -121,6 +121,7 @@ public void setWorker(final CitizenData citizen)
             {
                 citizen.resetExperienceAndLevel();
             }
+            tempCitizen.setLastJob(getJobName());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "tempCitizen" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2259)